src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -313,14 +313,10 @@ private static async Task<string> DownloadArtifactsAsync(BuildHttpClient buildCl
 
             using (Stream s = await buildClient.GetArtifactContentZipAsync(Options.TFSProjectName, build.Id, artifact.Name, cancellationToken))
             {
-                using (var fs = File.OpenWrite(archiveDownloadPath))
+                using (ZipArchive archive = new ZipArchive(s))
                 {
-                    // Using the default buffer size.
-                    await s.CopyToAsync(fs, 81920, cancellationToken);
+                    archive.ExtractToDirectory(tempDirectory);
```

Comment:
this change seems to make things a bit faster. instead of saving the whole zipfile to disk, we just stream the zip file into memory and then extract it to the destination folder.